webform_registration.inc

```diff
@@ -198,7 +198,7 @@ function webform_registration_submit($form, &$form_state) {
   //Add use field instances
   $user_field_info_instances = field_info_instances('user', 'user');
   foreach ($user_field_info_instances as $field_name => $instance) {
-    if (!in_array($fields[$field_name]['value'], _webform_registration_get_special_needles())) {
+    if (isset($fields[$field_name]) && !in_array($fields[$field_name]['value'], _webform_registration_get_special_needles())) {
```

Comment:
@argiepiano Would we be one step safer to use `is_array()` instead of `isset()` here?

Reply:
So, the issue that the above code was trying to solve was created by node entity cache problems. When you save the configuration found in `node/NID/webform/registration`, those settings include specific settings for each user field (whether the field should be omitted completely, or mapped onto a different webform component). The problem we encountered, addressed in #6, is that, unless you manually clear the caches after saving that form, Backdrop will use a stale version of the webform node that does not include these settings at all.

Therefore, when reaching this point in the code, there is no available setting for the user field. In this case, `$fields[$field_name]` is not set at all, so checking for `is_array()` will also throw a warning. And after clearing the cache, the code in  `_webform_registration_get_fields()` makes sure that `$fields[$field_name]` is always an array, so, again, checking for `is_array()` doesn't make sense there either.

After saying all of that, if you fix #6 by clearing the node entity cache after saving the webform's Registration Settings form, this fix here is not needed at all.